mshadow/tensor.h

```diff
@@ -1069,10 +1069,13 @@ inline void BatchGEMM(Tensor<Device, 3, DType> dst,
 #define MSHADOW_SCALAR_ double
 #include "./expr_scalar-inl.h"
 #undef MSHADOW_SCALAR_
-#define MSHADOW_SCALAR_ int
+#define MSHADOW_SCALAR_ int32_t
 #include "./expr_scalar-inl.h"
 #undef MSHADOW_SCALAR_
-#define MSHADOW_SCALAR_ mshadow::index_t
+#define MSHADOW_SCALAR_ uint32_t
```

Comment:
Is uint32_t used anywhere? Do we aim to support unsigned integers? I thought we deliberately change all unsigned ints to signed ints.